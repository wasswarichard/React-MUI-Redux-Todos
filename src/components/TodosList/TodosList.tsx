import React from 'react';
import TodoCard from "../TodoCard/TodoCard";
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import {ITODO} from "../../types/types";

const TodosList = ({todos, setStatusChange}: {todos: ITODO[], setStatusChange: any}) => {
    return (
        <div>
            <Card sx={{ maxWidth: 800 }} className="TodoList">
                <CardContent>
                    <div className="boardColumn-taskList">
                        { todos.length > 0 && todos.map( (todo : ITODO) => {
                            return(
                                <TodoCard key={todo.id} todo={todo} setStatusChange={setStatusChange} usage="listTodos"/>
                            )
                        } )}
                    </div>
                </CardContent>
            </Card>
        </div>
    );
}
export default TodosList;