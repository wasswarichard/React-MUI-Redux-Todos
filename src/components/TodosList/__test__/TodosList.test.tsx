import React from "react";
import TodosList from "../TodosList";
import Enzyme, { shallow } from 'enzyme';
import {cleanup} from "@testing-library/react";
import Adapter from 'enzyme-adapter-react-16';
import TestRenderer from 'react-test-renderer';
Enzyme.configure({ adapter: new Adapter() })

describe('<TodosList>', () => {

    afterEach(cleanup)
    const todoElements = [
        {userId: 1, id: 1, title: "DONE", completed: true, status: "DONE"},
        {userId: 2, id: 2, title: "DONE", completed: true, status: "DONE"}
    ]
    const statusChange = () => {};

    it('renders BoardColumn Component', () => {
        shallow(<TodosList todos={todoElements} setStatusChange={statusChange}/>);
    });

    it('should match snapshot', function () {
        const tree = TestRenderer.create(<TodosList todos={todoElements} setStatusChange={statusChange}/>).toJSON();
        expect(tree).toMatchSnapshot();

    });
});