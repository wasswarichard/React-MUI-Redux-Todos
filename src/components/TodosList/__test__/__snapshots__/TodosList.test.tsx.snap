// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodosList> should match snapshot 1`] = `
<div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root TodoList css-1nieira-MuiPaper-root-MuiCard-root"
  >
    <div
      className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        className="boardColumn-taskList"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root MuiButtonBase-root-card css-1hgxlf6-MuiPaper-root-MuiCard-root"
        >
          <div
            className="MuiAvatar-root MuiAvatar-circular MuiButtonBase-root-avatar"
          >
            <img
              alt="profile picture"
              className="MuiAvatar-img"
              src="anonymous.jpg"
            />
          </div>
          <div
            className="MuiButtonBase-root-content"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              1
              . 
              <span>
                DONE
              </span>
              <span
                style={
                  Object {
                    "marginLeft": "80px",
                  }
                }
              >
                DONE
              </span>
            </p>
          </div>
        </div>
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root MuiButtonBase-root-card css-1hgxlf6-MuiPaper-root-MuiCard-root"
        >
          <div
            className="MuiAvatar-root MuiAvatar-circular MuiButtonBase-root-avatar"
          >
            <img
              alt="profile picture"
              className="MuiAvatar-img"
              src="anonymous.jpg"
            />
          </div>
          <div
            className="MuiButtonBase-root-content"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              2
              . 
              <span>
                DONE
              </span>
              <span
                style={
                  Object {
                    "marginLeft": "80px",
                  }
                }
              >
                DONE
              </span>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
